lib/mixins/_base.js

```diff
@@ -5,8 +5,25 @@ const MIXIN = {
         config: {
             type: [ MantraData ],
             required: true,
+        },
+        role: {
+            type: String,
+            required: true,
+            validator: function(value) {
+                return ['ROOT', 'CHILD'].some(option => option === value);
+            }
```

Comment:
Just a little bit simpler 
```suggestion
            validator(value) {
                return ['ROOT', 'CHILD'].includes(value);
            }
```